Test fails if local bitcoind is running

Haven't dug much into this yet but the following test from #30 failed while I was running bitcoind locally  in `-regtest`, though I suspect the regtest part is irrelevant (my dev system points to my mainnet full node on a different machine; when running bitcoind locally it's only in regtest).

```
=================================================== test session starts ===================================================
platform darwin -- Python 3.7.2, pytest-5.2.2, py-1.8.0, pluggy-0.13.1
rootdir: /Users/kdmukai/dev/bitcoin/specter-desktop
collected 3 items                                                                                                         

tests/test_specter.py ..F                                                                                           [100%]

======================================================== FAILURES =========================================================
______________________________________________________ test_specter _______________________________________________________

specter_not_configured = <specter.Specter object at 0x105f5f8d0>

    def test_specter(specter_not_configured):
        specter_not_configured.check()
        assert specter_not_configured.wallets is not None
        assert specter_not_configured.devices is not None
        some_json = specter_not_configured.test_rpc()
>       assert some_json["out"] == ""
E       assert '{\n    "bloc...nings": ""\n}' == ''
E         - {
E         -     "blocks": 216,
E         -     "difficulty": 4.656542373906925e-10,
E         -     "networkhashps": 0.0009210894953580927,
E         -     "pooledtx": 1,
E         -     "chain": "regtest",
E         -     "warnings": ""...
E         
E         ...Full output truncated (2 lines hidden), use '-vv' to show

tests/test_specter.py:29: AssertionError
-------------------------------------------------- Captured stdout setup --------------------------------------------------
trying 1 different configs
Detected 1 bitcoin daemons
not loaded wallets: []
-------------------------------------------------- Captured stdout call ---------------------------------------------------
trying 1 different configs
Detected 1 bitcoin daemons
not loaded wallets: []
trying 1 different configs
Detected 1 bitcoin daemons
=============================================== 1 failed, 2 passed in 0.86s ===============================================
```

## Workaround
End the local bitcoind process and then the tests will pass once again.

--- a/requirements.txt
+++ b/requirements.txt
@@ -24,3 +24,4 @@ typing-extensions==3.7.4
 urllib3==1.25.6
 Werkzeug==0.16.0
 pytest==5.2.2
+python-bitcoinrpc==1.0

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -1,0 +1,66 @@
+import pytest
+import os
+
+
+import atexit
+import subprocess
+import tempfile
+import shutil
+import json
+import time
+#import unittest
+
+from bitcoinrpc.authproxy import AuthServiceProxy, JSONRPCException
+
+def start_bitcoind(bitcoind_path, cleanup_at_exit=True, rpc_port=18543):
+    ''' starts bitcoind with a specific rpcport=18543 by default.
+        That's not the standard in order to make pytest running while
+        developing locally against a different regtest-instance
+    '''
+    datadir = tempfile.mkdtemp()
+    bitcoind_proc = subprocess.Popen([
+        bitcoind_path, '-regtest', 
+        '-datadir=' + datadir, 
+        '-port=18544',
+        '-rpcport=' + str(rpc_port), 
+        '-noprinttoconsole'
+    ])
+    def cleanup_bitcoind():
+        bitcoind_proc.kill()
+        shutil.rmtree(datadir)
+    if cleanup_at_exit:
+        atexit.register(cleanup_bitcoind) 
+    # Wait for cookie file to be created
+    # ToDo: Why not acively setting rpc_username/password?!
+    while not os.path.exists(datadir + '/regtest/.cookie'):
+        time.sleep(0.5)
+    # Read .cookie file to get user and pass
+    with open(datadir + '/regtest/.cookie') as f:
+        rpc_username, rpc_password = f.readline().lstrip().rstrip().split(':')
+    rpc = AuthServiceProxy('http://{}:{}@127.0.0.1:{}/wallet/'.format(rpc_username, rpc_password, rpc_port))
+
+    # Wait for bitcoind to be ready
+    ready = False
+    while not ready:
+        try:
+            rpc.getblockchaininfo()
+            ready = True
+        except JSONRPCException:
+            time.sleep(0.5)
+            pass
+
+    # Make sure there are blocks and coins available
+    rpc.generatetoaddress(101, rpc.getnewaddress())
+    return (rpc, rpc_username, rpc_password, rpc_port)
+
+
+
+@pytest.fixture(scope="module")
+def bitcoin_regtest():
+    btc_conn = {}
+    #ToDo: compile our own version of bitcoind
+    #bitcoind_path = os.path.join(test_dir, 'bitcoin/src/bitcoind')
+    #bitcoind_path = '/usr/local/bin/bitcoind'
+    bitcoind_path = 'bitcoind' # Let's take the one on the path for now
+    btc_conn['rpc'], btc_conn['rpc_username'], btc_conn['rpc_password'], btc_conn['rpc_port'] = start_bitcoind(bitcoind_path)
+    return btc_conn

--- a/tests/helpers_testdata/some_jsonfile.json
+++ b/tests/helpers_testdata/some_jsonfile.json
@@ -1,0 +1,7 @@
+{
+    "blub":"bla",
+    "id":"ID123",
+    "deeper_key": {
+        "someotherkey":"someothervalue"
+    }
+}

--- a/tests/helpers_testdata/some_other_jsonfile.json
+++ b/tests/helpers_testdata/some_other_jsonfile.json
@@ -1,0 +1,7 @@
+{
+    "bla":"blub",
+    "id":"ID124",
+    "deeper_key": {
+        "someotherkey":"someothervalue"
+    }
+}

--- a/tests/test_helpers.py
+++ b/tests/test_helpers.py
@@ -1,0 +1,16 @@
+def test_load_jsons():
+    from specter import helpers
+    mydict = helpers.load_jsons("./tests/helpers_testdata")
+    assert mydict["some_jsonfile"]["blub"] == "bla"
+    assert mydict["some_other_jsonfile"]["bla"] == "blub"
+    mydict = helpers.load_jsons("./tests/helpers_testdata",'id')
+    assert "some_jsonfile" not in mydict
+    # instead the value for the key "id" is now used as the top-level key
+    assert mydict["ID123"]['blub'] == "bla"
+    # This also assumes that the key is unique!!!! 
+    assert mydict["ID124"]['bla'] == "blub"
+    # ToDo: check the uniqueness in the implementation to avoid issues
+    # the filename is added as alias
+    assert mydict["ID123"]['alias'] == "some_jsonfile"
+    # We also get the fullpath of that file:
+    assert mydict["ID123"]['fullpath'] == "./tests/helpers_testdata/some_jsonfile.json"

--- a/tests/test_specter.py
+++ b/tests/test_specter.py
@@ -1,14 +1,34 @@
-''' Tests for lovac.models '''
 import shutil
 import pytest
+import json
 from specter import Specter, alias
 
+
+
 @pytest.fixture
-def specter_not_configured():
+def empty_data_folder():
     # Make sure that this folder never ever gets a reasonable non-testing use-case
     data_folder = './test_specter_data_2789334'
     shutil.rmtree(data_folder, ignore_errors=True) 
-    yield Specter(data_folder=data_folder)    
+    yield data_folder
+    shutil.rmtree(data_folder, ignore_errors=True)
+
+@pytest.fixture
+def specter_regtest_configured(bitcoin_regtest):
+    # Make sure that this folder never ever gets a reasonable non-testing use-case
+    data_folder = './test_specter_data_3456778'
+    shutil.rmtree(data_folder, ignore_errors=True)
+    config = {
+        "rpc": {
+            "autodetect": False,
+            "user": bitcoin_regtest["rpc_username"],
+            "password": bitcoin_regtest["rpc_password"],
+            "port": bitcoin_regtest["rpc_port"],
+            "host": "localhost",
+            "protocol": "http"
+        },
+    }
+    yield Specter(data_folder=data_folder, config=config)
     shutil.rmtree(data_folder, ignore_errors=True)
 
 
@@ -17,14 +37,14 @@ def test_alias():
     assert alias("wurst_1") == "wurst_1"
     assert alias("Wurst$ 1") == "wurst_1"
 
-def test_specter_permrights():
-    with pytest.raises(Exception):
-        Specter("/notexisting_directory")
+def test_specter(specter_regtest_configured):
+    specter_regtest_configured.check()
+    assert specter_regtest_configured.wallets is not None
+    assert specter_regtest_configured.devices is not None
+    json_return = json.loads(specter_regtest_configured.test_rpc()["out"] )
+    assert json_return['blocks'] == 101
+    assert json_return['chain'] == 'regtest'
 
-def test_specter(specter_not_configured):
-    specter_not_configured.check()
-    assert specter_not_configured.wallets is not None
-    assert specter_not_configured.devices is not None
-    some_json = specter_not_configured.test_rpc()
-    assert some_json["out"] == ""
-    assert some_json["err"] == "autodetect failed"
+def test_device(empty_data_folder):
+    from specter import DeviceManager
+    my_dm = DeviceManager(data_folder=empty_data_folder)

